XCTest/XCTAssert.swift

```diff
@@ -10,13 +10,63 @@
 //  XCTAssert.swift
 //
 
-/**
-The primitive assertion function for XCTest. All other XCTAssert* functions are implemented in terms of this. This function emits a test failure if the general Bool expression passed to it evaluates to false.
-- Parameter expression: A boolean test. If it evaluates to false, the assertion fails and emits a test failure.
-- Parameter message: An optional message to use in the failure if the assetion fails. If no message is supplied a default message is used.
-- Parameter file: The file name to use in the error message if the assertion fails. Default is the file containing the call to this function. It is rare to provide this parameter when calling this function.
-- Parameter line: The line number to use in the error message if the assertion fails. Default is the line number of the call to this function in the calling file. It is rare to provide this parameter when calling this function.
-*/
+/// The primitive assertion function for XCTest. All other XCTAssert* functions
+/// are implemented in terms of this. This function emits a test failure if the
+/// general Bool expression passed to it evaluates to false.
+///
+/// - Requires: This and all other XCTAssert* functions must be called from
+///   within a test method, as indicated by `XCTestCaseProvider.allTests`.
+///   Assertion failures that occur outside of a test method will *not* be
+///   reported as failures.
+///
+/// - Parameter expression: A boolean test. If it evaluates to false, the
+///   assertion fails and emits a test failure.
+/// - Parameter message: An optional message to use in the failure if the
```

Comment:
This change conflicts with #22--@briancroom and I both fix the `assetion` typo :blush: